xdsl/dialects/riscv.py

```diff
@@ -1249,6 +1249,44 @@ class EbreakOp(NullaryOperation):
 
 # endregion
 
+# RISC-V SSA Helpers
+
+
+@irdl_op_definition
+class GetRegisterOp(IRDLOperation, RISCVOp):
+    """
+    This instruction allows us to create an SSAValue with for a given register name. This
+    is useful for bridging the RISC-V convention that stores the result of function calls
+    in `a0` and `a1` into SSA form.
+
+    For example, to generate this assembly:
+    ```
+    jal my_func
+    add a0 s0 a0
+    ```
+
+    One needs to do the following:
+
+    ``` python
+    rhs = riscv.GetRegisterOp(Registers.s0).res
+    riscv.JalOp("my_func")
+    lhs = riscv.GetRegisterOp(Registers.A0).res
+    sum = riscv.AddOp(lhs, rhs, Registers.A0).rd
+    ```
+    """
+
+    name = "riscv.get_register"
```

Comment:
maybe we can do something like `riscv.mlir.get_register`? MLIR does that in their llvm dialect with things that are not part of llvm, but were introduced to enable using it with MLIR. I feel like this is an ssa-specific op that is not "like" the other operations in this dialect, in that it does not represent a part of the target...

Reply:
So in my mind the thing that defined the `riscv` dialect is that it should only include operations that should be present during assembly printing. Since this operation is required for that, I thought I'd put it here.

Reply:
So it makes sense for it to be a RISCVOp and a part of the riscv dialect, and then I'm not sure what semantically `riscv.mlir.bla` means in the op name wrt the dialect structure, my impression is that it's a single level construct both in xDSL and MLIR, and that multiple periods aren't an official thing, so might be a bit more misleading

Reply:
Oh, I would still put it in the riscv dialect, I would just suggest the name to be `riscv.<something>.get_register`.

Reply:
I'll merge as-is, happy to rename, would like to open another PR that depends on this. Maybe the solution would be to move this op and some others like it to a `riscv_assembly` dialect